dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -34,33 +36,119 @@
 import org.apache.bcel.classfile.InnerClass;
 import org.apache.bcel.classfile.InnerClasses;
 import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
 import org.apache.bcel.util.ClassPath;
 import org.apache.bcel.util.SyntheticRepository;
 
 /**
- * This class reads a jar file and lists method references in its class files.
+ * This class reads jar files and runs static linkage analysis on them
  */
-class JarDumper {
+class StaticLinkageChecker {
+
+  /**
+   * Given list of the jar files as file names in filesystem, outputs the report of static linkage
+   * check.
+   *
+   * @param arguments jar files available in the file system
+   * @throws IOException when there is a problem in reading a jar file
+   * @throws ClassNotFoundException when there is a problem in reading a class from a jar file
+   */
+  public static void main(String[] arguments) throws IOException, ClassNotFoundException {
+    String report = generateStaticLinkageReport(arguments);
+    System.out.println(report);
+  }
+
+  @VisibleForTesting
+  static String generateStaticLinkageReport(String[] jarFileNames) throws IOException,
+      ClassNotFoundException {
+    StringBuilder stringBuilder = new StringBuilder();
+    Set<FullyQualifiedMethodSignature> externalMethodReferences = new HashSet<>();
+    List<Path> paths = new ArrayList<>();
+    for (String jarFileName : jarFileNames) {
+      File jarFile = new File(jarFileName);
+      Path jarFilePath = jarFile.toPath();
+      paths.add(jarFilePath);
+      externalMethodReferences.addAll(listExternalMethodReferences(jarFilePath));
+    }
+    List<FullyQualifiedMethodSignature> unresolvedMethodReferences =
+        resolvedMethodReferences(paths, Lists.newArrayList(externalMethodReferences));
+    if (unresolvedMethodReferences.isEmpty()) {
+      stringBuilder.append("There were no unresolved method references from the jar file(s) :");
+      stringBuilder.append(paths);
+    } else {
+      stringBuilder.append("There were unresolved method references from the jar file(s):\n");
+      for (FullyQualifiedMethodSignature methodReference : unresolvedMethodReferences) {
+        stringBuilder.append("  ");
+        stringBuilder.append(methodReference);
+        stringBuilder.append("\n");
+      }
+    }
+    return stringBuilder.toString();
+  }
+
+  /**
+   * Checks the availability of the methods through the jar files, and lists the unavailable
+   * methods.
+   *
+   * @param paths jar files to search for the availability of the methods
+   * @param methodReferences methods to search for with the jar files
+   * @return list of methods that are not found in the jar files
+   */
+  static List<FullyQualifiedMethodSignature> resolvedMethodReferences(List<Path> paths,
```

Comment:
The content that is returned doesn't match the method name - the name implies it returns the references that were resolved, but it actually returns the opposite. I suggest `findUnresolvedReferences`.